core/src/bson_type_info.rs

```diff
@@ -0,0 +1,180 @@
+#[non_exhaustive]
+pub struct BsonTypeInfo {
```

Comment:
This is not an exact copy of the `BsonTypeInfo` we have in the JDBC driver; this only contains the relevant fields needed to populate `MongoColMetadata`. This can be easily extended in the future to contain more or less information.

I wasn't sure the best way to represent this type of data in Rust. I went with a non-exhaustive struct -- so no other variants can be made except those we construct in this crate (I think 🤔 ) -- and a ` const` instance of the struct for each BSON type. Some alternatives ideas I had were creating a `BsonTypeInfo` enum with each BSON type being a variant and having an `impl` with methods like `get_type_name` and `get_is_searchable` that would match on each variant and return the appropriate value. This seemed like it would be a lot more cumbersome though since each method would have the full exhaustive pattern match. LMK if there are other alternatives that could be better or at least more idiomatic.